src/Symfony/Component/Messenger/EventListener/SendFailedMessageToFailureTransportListener.php

```diff
@@ -27,12 +28,17 @@
  */
 class SendFailedMessageToFailureTransportListener implements EventSubscriberInterface
 {
-    private $failureSender;
+    private $failureSenders;
     private $logger;
+    /**
+     * @var SenderInterface|null global failure sender will be null if it is not defined
+     */
+    private $globalfailureSender;
 
-    public function __construct(SenderInterface $failureSender, LoggerInterface $logger = null)
+    public function __construct(SenderInterface $globalfailureSender = null, LoggerInterface $logger = null, ServiceLocator $failureSenders = null)
```

Comment:
I would use the same strategy here that I mentioned for the `AbstractFailedMesagesCommand`: do NOT add a new argument, but instead, remove the type-hint from this first argument, trigger a deprecation warning of the old value is passed, and write code to make the old code still work.

Reply:
@weaverryan  I need to get the transport name from a SenderInterface. How can I get that? I need it, because in the event, we get the receiver name and get from the locator the sender to send to the failed transport.